### PR TITLE
Scroll prompt buffer entries with mouse only when focused

### PR DESCRIPTION
**Pull request by Ambrevar**
This is a natural follow-up to #1392 and finally a fix to #399.

### CODE REVIEW

**Comment by jmercouris** [comment]
Ah, so the user will now be able to click and scroll, instead of hijacking all of the scroll because of the addition of the `is-focused` generic function!

File: source/input.lisp [hunk 1]
```diff
@@ -32,9 +32,10 @@ Example:
     keymap))
 
 (export-always 'current-keymaps)
-(defun current-keymaps (&optional (buffer (if (active-prompt-buffers (current-window))
-                                              (current-prompt-buffer)
-                                              (current-buffer))))
+(defun current-keymaps (&optional (buffer (let ((prompt-buffer (current-prompt-buffer)))
```
Comment (jmercouris) [inline source/input.lisp:35]: Why not generalize this and just retrieve the focused buffer from the window? Have a method of the window `focused-buffer` or something like that?

Comment (jmercouris) [reply source/input.lisp:35]: Note that this change would also be very helpful when we have buffer splitting.

Comment (jmercouris) [reply source/input.lisp:35]: On the other hand, it adds extra complexity to this PR, and we still have `active-buffer`, which would need to be dealt with. I am thinking this does not belong in this PR, and that this technique suffices.

Comment (Ambrevar) [reply source/input.lisp:35]: All good points!

**Pull request merged**